gazebo_plugins/src/gazebo_ros_bumper.cpp

```diff
@@ -0,0 +1,226 @@
+// Copyright 2019 Open Source Robotics Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+ * \desc Bumper controller
+ * \author Nate Koenig
+ * \date 09 Sept. 2008
+ */
+
+#include <gazebo/plugins/ContactPlugin.hh>
+
+#include <gazebo_msgs/msg/contact_state.hpp>
+#include <gazebo_msgs/msg/contacts_state.hpp>
+#include <gazebo_plugins/gazebo_ros_bumper.hpp>
+#include <gazebo_ros/conversions/builtin_interfaces.hpp>
+#include <gazebo_ros/node.hpp>
+#include <gazebo_ros/utils.hpp>
+#include <geometry_msgs/msg/wrench.hpp>
+
+#include <memory>
+#include <string>
+
+namespace gazebo_plugins
+{
+class GazeboRosBumperPrivate
+{
+public:
+  /// Callback to be called on contact.
```

Comment:
Actually this is being called whenever the sensor is updated, regardless of whether there has been a contact or not.